lib/util/http/host-rules.ts

```diff
@@ -162,6 +164,27 @@ export function applyHostRule<GotOptions extends HostRulesGotOptions>(
     options.lookup = dnsLookup;
   }
 
+  if (hostRule.headers) {
+    const allowedHeaders = GlobalConfig.get('allowedHeaders');
+    const filteredHeaders: { [key: string]: string } = {};
+
+    for (const [header, value] of Object.entries(hostRule.headers)) {
+      if (anyMatchRegexOrMinimatch(allowedHeaders, header)) {
+        filteredHeaders[header] = value;
+      } else {
+        logger.once.warn(
+          { allowedHeaders, header },
+          'Disallowed hostRules header',
+        );
+      }
+    }
+
+    options.headers = {
+      ...options.headers,
+      ...filteredHeaders,
```

Comment:
I would prefer to reverse order, so our headers are preferred and not overridden